diff --git a/modules/v2/ar.py b/modules/v2/ar.py
index 0611a45..1bf16a1 100644
--- a/modules/v2/ar.py
+++ b/modules/v2/ar.py
@@ -384,6 +384,7 @@ class NaiveWrapper(nn.Module):
             prompt_text,
             prompt_target,
             compiled_decode_fn = None,
+            min_tokens = 0,
             **sampling_kwargs,
     ):
         sep_token_emb = self.sep_token_emb.expand(1, 1, -1)
@@ -402,7 +403,7 @@ class NaiveWrapper(nn.Module):
         new_emb = self.model.embed_base(pred_base.unsqueeze(0), torch.LongTensor([1]).to(pred_base.device))[1]
         emb_seq = torch.cat([emb_seq, new_emb], dim=1)
         for _ in tqdm(range(4000)):
-            suppress_eos = len(pred_codes) < 10
+            suppress_eos = len(pred_codes) < 10 or len(pred_codes) < min_tokens
             input_pos = input_pos[-1:] + 1
             kv_pos = kv_pos[-1:] + 1
             next_tokens = self.decode_one_token_ar(
